framework/src/media/audio/audio_manager.c

```diff
@@ -155,21 +194,54 @@ audio_manager_result_t find_output_audio_card(void)
 	}
 
 	while ((dir_entry = readdir(dir_info)) != NULL) {
+		char path[256] = "/dev/audio/";
 		audvdbg("Found a directory %s\n", dir_entry->d_name);
 		// TODO: Add cases for various drivers. Currently, identify 'pcm' drivers only.
 		if ((dir_entry->d_name[0] != 'p')
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
+		strcat(path, dir_entry->d_name);
 		if ((type == 'p') && (card_id < CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
+			int fd;
+			struct audio_caps_desc_s caps_desc;
+			FAR struct audio_caps_s *caps = &(caps_desc.caps);
+
+			if ((fd = open(path, O_WRONLY)) < 0) {
+				auddbg("Fail to open output driver, path = %s\n", path);
+				return AUDIO_MANAGER_DEVICE_FAIL;
+			}
+
+			caps->ac_len = sizeof(struct audio_caps_s);
+			caps->ac_type = AUDIO_TYPE_FEATURE;
+			caps->ac_format.hw = AUDIO_FU_VOLUME;
+
+			if ((ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long)caps)) < 0) {
+				auddbg("Fail to ioctl AUDIOIOC_GETCAPS, ret = %d\n", ret);
+				close(fd);
+				return AUDIO_MANAGER_DEVICE_FAIL;
+			}
+
+			g_audio_out_cards[card_id].config.max_volume = caps->ac_controls.b[0];
+			g_audio_out_cards[card_id].config.volume = caps->ac_controls.b[1] * 10 / caps->ac_controls.b[0];
+			close(fd);
+#endif
+			g_audio_out_cards[card_id].card_path = (char *)malloc(sizeof(path) + 1);
```

Comment:
First set path here, and use get_in/out_audio_volume what you added.

Reply:
Okay. Modified.